sequence.go

```diff
@@ -9,14 +9,20 @@ package goscale
 
 import (
 	"bytes"
+	"reflect"
 )
 
 type Sequence[T Encodable] []T
 
 func (seq Sequence[Encodable]) Encode(buffer *bytes.Buffer) {
 	Compact(len(seq)).Encode(buffer)
+
 	for _, v := range seq {
-		v.Encode(buffer)
+		if reflect.TypeOf(v).Kind() != reflect.Struct {
```

Comment:
Could we revert the switch? I think it would be more readable.

Reply:
👍